Frameworks/Foundation/NSString.mm

```diff
@@ -680,7 +680,25 @@ - (instancetype)pathExtension {
  @Status Interoperable
 */
 - (instancetype)lastPathComponent {
-    return [self substringFromIndex:_CFStartOfLastPathComponent2(static_cast<CFStringRef>(self))];
+    if ([self length] == 0) {
+        return @"";
+    }
+
+    CFIndex index = _CFStartOfLastPathComponent2(static_cast<CFStringRef>(self));
+    NSString* substring = [self substringFromIndex:index];
+
+    if (IS_SLASH([self characterAtIndex:index])) {
+        if ([self length] == 1) {
+            return substring;
+        }
+
+        // only way we could have ended up here is, if we have a string with multiple leading slashes.
+        NSArray<NSString*>* tokens = [self componentsSeparatedByString:_NSGetSlashStr()];
+        return ([tokens[0] isEqualToString:@""]) ? _NSGetSlashStr() : tokens[0];
```

Comment:
>tokens[0] [](start = 17, length = 9)

blind index here seems risky? Any concerns for that / comments we need to reassure the reader that this is ok?